webservices/rest.py

```diff
@@ -223,65 +162,21 @@ def add_caching_headers(response):
             'public, max-age={}'.format(app.config['MAX_CACHE_AGE'])
         )
 
-    if (is_cacheable_endpoint(response.status_code, request.path)):
-        # Convert the response content into a JSON object and format the URL by
-        # removing the api_key parameter and other special characters.
-        json_data = utils.get_json_data(response)
-        formatted_url = utils.format_url(request.url)
-
-        app.logger.info('Attempting to cache request at: {}'.format(request.url))
-        cache_request.cache_all_requests.delay(json_data, formatted_url)
-
     return response
 
 
 @app.errorhandler(Exception)
 def handle_exception(exception):
     wrapped = ResponseException(str(exception), ErrorCode.INTERNAL_ERROR, type(exception))
 
-    app.logger.info(
+    app.logger.error(
         'An API error occurred with the status code of {status} ({exception}).'
         .format(
             status=wrapped.status,
             exception=wrapped.wrappedException
         )
     )
 
-    if is_retrievable_from_cache(wrapped.status, request.path):
-        app.logger.info('Attempting to retrieving the cached request from S3...')
-
-        # Retrieve the information needed to construct a URL for the S3 bucket
-        # where the cached API responses live.
-        formatted_url = utils.format_url(request.url)
-        s3_bucket = utils.get_bucket()
-        bucket_region = env.get_credential('region')
-        cached_url = "http://s3-{0}.amazonaws.com/{1}/cached-calls/{2}".format(
-            bucket_region,
-            s3_bucket.name,
-            formatted_url
-        )
-
-        # Attempt to retrieve the cached data from S3.
-        cached_data = utils.get_cached_request(cached_url)
-
-        # If the cached data was returned, we can return that to the client.
-        # Otherwise, log the error and raise an API error.
-        if cached_data is not None:
-            app.logger.info('Successfully retrieved cached request from S3.')
-            return cached_data
-        else:
-            app.logger.error(
-                'An error occured while retrieving the cached file from S3.'
-            )
-            raise exceptions.ApiError(
-                'The requested URL could not be found.'.format(request.url),
-                status_code=http.client.NOT_FOUND
-            )
-    else:
-        raise exceptions.ApiError(
```

Comment:
We should still raise this exception if we cannot return a response. We may need to revisit exception handling so that we are in control of what we return to the client.